hooks/useFuturesData.ts

```diff
@@ -0,0 +1,272 @@
+				: marketSize.sub(marketSkew).div('2').abs().mul(marketAssetRate).gte(maxMarketValueUSD),
+		[leverageSide, marketSize, marketSkew, marketAssetRate, maxMarketValueUSD]
+	);
+
+	const onTradeAmountChange = React.useCallback(
+		(value: string, fromLeverage: boolean = false) => {
+			const size = fromLeverage ? (value === '' ? '' : wei(value).toNumber().toString()) : value;
+			const sizeSUSD = value === '' ? '' : marketAssetRate.mul(Number(value)).toNumber().toString();
+			const leverage =
+				value === ''
+					? ''
+					: marketAssetRate
+							.mul(Number(value))
+							.div(position?.remainingMargin)
+							.toString()
+							.substring(0, 4);
+			setTradeSize(size);
+			setTradeSizeSUSD(sizeSUSD);
+			setLeverage(leverage);
+		},
+		[marketAssetRate, position?.remainingMargin, setTradeSize, setTradeSizeSUSD, setLeverage]
+	);
+
+	useEffect(() => {
+		const handleRouteChange = () => {
+			setTradeSize('');
+			setTradeSizeSUSD('');
+			setLeverage('');
+		};
+
+		router.events.on('routeChangeStart', handleRouteChange);
+
+		return () => {
+			router.events.off('routeChangeStart', handleRouteChange);
+		};
+	}, [router.events, setTradeSize, setTradeSizeSUSD, setLeverage]);
+
+	const onTradeAmountSUSDChange = (value: string) => {
+		const valueIsNull = value === '' || Number(value) === 0;
+		if (marketAssetRate.gt(0)) {
+			const size = valueIsNull ? '' : wei(value).div(marketAssetRate).toNumber().toString();
+			const leverage = valueIsNull
+				? ''
+				: wei(value).div(position?.remainingMargin).toString().substring(0, 4);
+			setTradeSizeSUSD(value);
+			setTradeSize(size);
+			setLeverage(leverage);
+		}
+	};
+
+	const onLeverageChange = React.useCallback(
+		(value: string) => {
+			if (value === '' || Number(value) <= 0) {
+				setTradeSize('');
+				setTradeSizeSUSD('');
+				setLeverage(Number(value) === 0 ? value.substring(0, 4) : '');
+			} else {
+				const newTradeSize = marketAssetRate.eq(0)
+					? 0
+					: wei(value)
+							.mul(position?.remainingMargin ?? zeroBN)
+							.div(marketAssetRate);
+				onTradeAmountChange(newTradeSize.toString(), true);
+				setLeverage(value.substring(0, 4));
+			}
+		},
+		[
+			position?.remainingMargin,
+			marketAssetRate,
+			onTradeAmountChange,
+			setTradeSize,
+			setTradeSizeSUSD,
+			setLeverage,
+		]
+	);
+
+	const orderTxn = useSynthetixTxn(
+		`FuturesMarket${marketAsset?.[0] === 's' ? marketAsset?.substring(1) : marketAsset}`,
+		orderType === 1 ? 'submitNextPriceOrderWithTracking' : 'modifyPositionWithTracking',
+		[sizeDelta.toBN(), KWENTA_TRACKING_CODE],
+		gasPrice,
+		{
+			enabled:
+				!!marketAsset &&
+				!!leverage &&
+				Number(leverage) >= 0 &&
+				maxLeverageValue.gte(leverage) &&
+				!sizeDelta.eq(zeroBN),
+		}
+	);
+
+	useEffect(() => {
+		if (orderTxn.hash) {
+			monitorTransaction({
+				txHash: orderTxn.hash,
+				onTxConfirmed: () => {
+					onLeverageChange('');
+					handleRefetch('modify-position');
```

Comment:
Yeah. I think that's out of the scope of this PR though. I can add it to the list of things that we need to add in a separate PR.

Reply:
It's a bit trickier to do here because I believe some of the queries we refetch here depend on subgraph data, as opposed to raw contract data.

Reply:
I'm not sure it would make a difference if contract data or subgraph data, it would just be the query listening to a contract event to trigger a refresh, rather than the refresh being called after each transaction.